cpp/harvest/nacjd.cc

```diff
@@ -158,26 +165,52 @@ void ExtractExistingIDsFromMarc(MARC::Reader * const marc_reader, std::set<std::
     }
 }
 
+void ExtractOneIDFromWebsite(const unsigned nacjd_id) {
+    std::string nacjd_title_path("/tmp/nacjd_titles.html");
+    std::string nacjd_JSON_path("/tmp/nacjd_new_titles.json");
+
+    if (FileUtil::Exists(nacjd_title_path))
+        FileUtil::DeleteFile(nacjd_title_path);
+
+    FileUtil::WriteString(nacjd_title_path, "");
+
+    std::ifstream file(nacjd_title_path);
+    if (file)
+        std::for_each(std::istream_iterator<char>(file), std::istream_iterator<char>(), HandleChar);
+    else
+        LOG_ERROR("couldn't open file: " + nacjd_title_path);
+    if (FileUtil::Exists(nacjd_JSON_path) and not FileUtil::DeleteFile(nacjd_JSON_path))
+        LOG_ERROR("Could not delete file: " + nacjd_JSON_path);
+
+    std::ofstream json_new_titles(nacjd_JSON_path);
+
+    json_new_titles << "{ \"nacjd\" : [ " << '\n';
+    bool first(true);
+
+    DownloadID(json_new_titles, std::to_string(nacjd_id), /*use_separator*/ not first);
+
+    json_new_titles << " ] }";
+}
 
 void ExtractIDsFromWebsite(const std::set<std::string> &parsed_marc_ids, unsigned * const number_of_new_ids, const unsigned count_rows) {
     const std::string DOWNLOAD_URL(
         "https://www.icpsr.umich.edu/web/NACJD/search/"
         "studies?start=0&ARCHIVE=NACJD&PUBLISH_STATUS=PUBLISHED&sort=DATEUPDATED%20desc&rows="
         + std::to_string(count_rows));
-    if (FileUtil::Exists(NACJD_TITLES))
-        FileUtil::DeleteFile(NACJD_TITLES);
+    if (FileUtil::Exists("/tmp/nacjd_titles.html"))
```

Comment:
This HTML file should only be used inside this function. Instead of hardcoding a path, please use FileUtil::AutoTempFile instead. It will automatically generate a temporary file in /tmp and automatically delete it as soon as the object goes out of scope (e.g. when the function terminates successfully).